src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -889,6 +889,48 @@
     end
   end
 
+  describe 'POST #save_prjconf' do
+    before do
+      login user
+    end
+
+    context 'can save a project config' do
+      before do
+        post :save_prjconf, params: { project: user.home_project.name, submit: 'config', config: 'save config' }
+      end
+
+      it { expect(flash[:success]).to eq('Config successfully saved!') }
+      it { expect(response.status).to eq(200) }
+    end
+
+    context 'cannot save a project config' do
+      before do
+        allow_any_instance_of(ProjectConfigFile).to receive(:save).and_return(nil)
+        post :save_prjconf, params: { project: user.home_project.name, submit: 'config', config: '' }
+      end
+
+      it { expect(flash[:error]).not_to eq(nil) }
```

Comment:
this can written like this

`expect(flash[:error]).not_to be_nil`

However, I would check that the error messages is equal to the error you get.